presto-spark-base/src/test/java/com/facebook/presto/spark/PrestoSparkQueryRunner.java

```diff
@@ -627,6 +648,7 @@ public SparkContext get()
                             .set("spark.driver.host", "localhost")
                             .set(SPARK_EXECUTOR_CORES_PROPERTY, "4")
                             .set(SPARK_TASK_CPUS_PROPERTY, "4");
+                    additionalSparkConfigs.forEach(sparkConfiguration::set);
```

Comment:
Could you explain a bit more about why we need 
resetSparkContext(Map<String, String> additionalSparkConfigs)
get(Map<String, String> additionalSparkConfigs)
My understanding is you want add additional configs to existing sparkContext? And this is only needed for testing 
Is this correct or not?

Reply:
looks good overall, have one question about changes in PrestoSparkQueryRunner

Reply:
The reason we need reset: the Spark shuffle manager will be only created/initialized once at Spark context/Env creation time, and it'll be used throughout the life time of that Spark context. On other side, In PrestoSpark test suite, all the test cases in one suite will be sharing one Spark context (held by the PrestoSparkQueryRunner) by default.  With these two constraints, we have to 'reset' the Spark Context to create a new one to register our new shuffle manager for our testing purpose.